### PR TITLE
feat: change name on User to first + last, update tests and UI UN-137

### PR DESCRIPTION
**Pull request by PThorpe92**
image

This adds a first + last name to every UI, DB table, migration, test, factory and ensures all of them test and display properly :+1:

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -13,7 +13,8 @@ public function up(): void
     {
         Schema::create('users', function (Blueprint $table) {
             $table->id();
-            $table->string('name');
+            $table->string('name_first');
```
Comment (nokierae) [inline database/migrations/2014_10_12_000000_create_users_table.php:16]: We don't want to modify old migrations. We need a new migration that drops the name column, and adds the new columns.

File: database/migrations/2019_12_14_000001_create_personal_access_tokens_table.php [hunk 2]
```diff
@@ -14,7 +14,8 @@ public function up(): void
         Schema::create('personal_access_tokens', function (Blueprint $table) {
             $table->id();
             $table->morphs('tokenable');
-            $table->string('name');
+            $table->string('name_first');
```
Comment (nokierae) [inline database/migrations/2019_12_14_000001_create_personal_access_tokens_table.php:17]: See above.

File: resources/js/Layouts/AuthenticatedLayout.tsx [hunk 3]
```diff
@@ -7,122 +7,122 @@ import { Link } from '@inertiajs/react';
...
-                                            >
-                                                {user.name}
+            <div className="hidden sm:flex sm:items-center sm:ms-6">
+              <div className="ms-3 relative">
+                <Dropdown>
+                  <Dropdown.Trigger>
+                    <span className="inline-flex rounded-md">
+                      <button
+                        type="button"
+                        className="inline-flex items-center px-3 py-2 border border-transparent text-sm leading-4 font-medium rounded-md text-gray-500 dark:text-gray-400 bg-white dark:bg-gray-800 hover:text-gray-700 dark:hover:text-gray-300 focus:outline-none transition ease-in-out duration-150"
```
Comment (nokierae) [inline resources/js/Layouts/AuthenticatedLayout.tsx:38]: Lot's of changes from slate to gray here.

File: resources/js/Layouts/AuthenticatedLayout.tsx [hunk 4]
```diff
@@ -7,122 +7,122 @@ import { Link } from '@inertiajs/react';
...
+                    </Dropdown.Link>
+                  </Dropdown.Content>
+                </Dropdown>
+              </div>
+            </div>
 
-                        <div className="-me-2 flex items-center sm:hidden">
-                            <button
-                                onClick={() => setShowingNavigationDropdown((previousState) => !previousState)}
-                                className="inline-flex items-center justify-center p-2 rounded-md text-slate-400 dark:text-slate-500 hover:text-slate-500 dark:hover:text-slate-400 hover:bg-slate-100 dark:hover:bg-slate-900 focus:outline-none focus:bg-slate-100 dark:focus:bg-slate-900 focus:text-slate-500 dark:focus:text-slate-400 transition duration-150 ease-in-out"
```
Comment (nokierae) [inline resources/js/Layouts/AuthenticatedLayout.tsx:71]: Lot's of changes from slate to gray here.

Comment (nokierae) [reply resources/js/Layouts/AuthenticatedLayout.tsx:71]: Actually, I see it in a lot of the files below as well if you could swap them back.

Comment (PThorpe92) [reply resources/js/Layouts/AuthenticatedLayout.tsx:71]: Ahhh must have been from the merge conflict :/

File: database/migrations/2023_11_29_231426_users_drop_name.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::table('users', function (Blueprint $table) {
+            // Reverse the changes if needed
+            $table->string('name');
+            $table->dropColumn('name_first');
```
[the remark this answers is not in the document]
Comment (PThorpe92) [reply database/migrations/2023_11_29_231426_users_drop_name.php:27]: Yup good catch

**Review by nokierae — APPROVED** [review]

**Pull request merged**